File: Solutions/881. Boats to Save People/881.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int start = 0, end = people.length - 1, ans = 0;
        while(start <= end){
            if(people[start] + people[end] <= limit){
                start++;
                end--;
                ans++;
            } else{
                if(people[end] <= limit){
                    end--;
                    ans++;
                }
                else if(people[start] <= limit){
                    start++;
                    ans++;
                }
            }
        }
        return ans;
    }
}

File: Solutions/881. Boats to Save People/881.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        ans = 0
        start = 0
        end = len(people) - 1
        while start <= end:
            if people[start] + people[end] <= limit:
                start += 1
                end -= 1
                ans += 1
            else:
                if people[end] <= limit:
                    end -= 1
                    ans += 1
                elif people[start] <= limit:
                    start += 1
                    ans += 1
        return ans